Tests/M5Core2TestApp/Program.cs

```diff
@@ -1,94 +1,110 @@
-    if (WifiNetworkHelper.HelperException != null)
-    {
-        Debug.WriteLine($"ex: {WifiNetworkHelper.HelperException}");
-    }
-}
 
-var mpu = M5Core2.AccelerometerGyroscope;
+var mpu6886 = M5Core2.AccelerometerGyroscope;
 
 M5Core2.TouchEvent += TouchEventCallback;
 
+Console.ForegroundColor = nanoFramework.Presentation.Media.Color.Green;
+Console.WriteLine("Press the display or a button to get started!");
+Console.ForegroundColor = nanoFramework.Presentation.Media.Color.White;
+
 Thread.Sleep(Timeout.Infinite);
 
 void TouchEventCallback(object sender, TouchEventArgs e)
 {
-    const string StrLB = "LEFT BUTTON PRESSED  ";
-    const string StrMB = "MIDDLE BUTTON PRESSED  ";
-    const string StrRB = "RIGHT BUTTON PRESSED  ";
-    const string StrXY1 = "TOUCHED at X= ";
-    const string StrXY2 = ",Y= ";
-    const string StrID = ",Id= ";
-    const string StrDoubleTouch = "Double touch. ";
-    const string StrMove = "Moving... ";
-    const string StrLiftUp = "Lift up. ";
+    Console.Clear();
 
     Debug.WriteLine($"Touch Panel Event Received Category= {e.EventCategory} Subcategory= {e.TouchEventCategory}");
     Console.CursorLeft = 0;
     Console.CursorTop = 0;
 
-    Debug.WriteLine(StrXY1 + e.X + StrXY2 + e.Y + StrID + e.Id);
-    Console.WriteLine(StrXY1 + e.X + StrXY2 + e.Y + StrID + e.Id + "  ");
+    var eventMsg = $"{StrTouchEvent}({e.X},{e.Y}){StrEventID}{e.Id}";
+    Debug.WriteLine(eventMsg);
+    Console.WriteLine(eventMsg);
 
     if ((e.TouchEventCategory & TouchEventCategory.LeftButton) == TouchEventCategory.LeftButton)
     {
+        Vibrate(150);
         Debug.WriteLine(StrLB);
         Console.WriteLine(StrLB);
     }
     else if ((e.TouchEventCategory & TouchEventCategory.MiddleButton) == TouchEventCategory.MiddleButton)
     {
+        Vibrate(150);
         Debug.WriteLine(StrMB);
         Console.WriteLine(StrMB);
     }
     else if ((e.TouchEventCategory & TouchEventCategory.RightButton) == TouchEventCategory.RightButton)
     {
+        Vibrate(150);
         Debug.WriteLine(StrRB);
         Console.WriteLine(StrRB);
     }
 
     if ((e.TouchEventCategory & TouchEventCategory.Moving) == TouchEventCategory.Moving)
     {
-        Debug.WriteLine(StrMove);
-        Console.Write(StrMove);
+        Debug.WriteLine($"{StrTouchEventCatagory}{StrMove}");
+        Console.WriteLine($"{StrTouchEventCatagory}{StrMove}");
     }
 
     if ((e.TouchEventCategory & TouchEventCategory.LiftUp) == TouchEventCategory.LiftUp)
     {
-        Debug.WriteLine(StrLiftUp);
-        Console.Write(StrLiftUp);
+        Debug.WriteLine($"{StrTouchEventCatagory}{StrLiftUp}");
+        Console.WriteLine($"{StrTouchEventCatagory}{StrLiftUp}");
     }
 
-    if ((e.TouchEventCategory & TouchEventCategory.DoubleTouch) == TouchEventCategory.DoubleTouch)
+    if ((e.TouchEventCategory & TouchEventCategory.DoubleTouch) == TouchEventCategory.DoubleTouch) //TODO: never seems to fire!
```

Comment:
This is a limitation of the touch controller used in the Core2: it doesn't support double tap. So this code can be removed.

Reply:
Improved the comment.